Makefile

```diff
@@ -1,5 +1,118 @@
 # Citus toplevel Makefile
 
+EXTENSION = citus
+DATA = citus--7.0-15.sql
+
+
+MODULE_big = citus
+
+OBJS = src/backend/distributed/shared_library_init.o \
```

Comment:
Hey so this is crazy but what if we generate the Makefile using a minimal Docker image we run on Windows?

Reply:
IInteresting. This goes to something I was thinking about yesterday when mulling over `citus_version.h`, I wonder how portable we want this to be. I mean, I can get appveyor running perfectly by installing `autoconf` and `make` and making small changes to our existing makefiles, but I don't know how much help I should be giving other people who might want to build Citus in the future.

Presumably Docker is in the same category. Should we require anyone building Citus on Windows to have docker?

Reply:
I'm not saying to make users install it; maybe we should just check in `.win` versions of these files or something. I'm just saying for the purposes of making a script _we_ can run when we need to to update various files, it might be nice to script it somewhere we're comfortable with tools (i.e. Linux) and just run it against a shared-mount folder when we need to update these things in Windows.